src/mappings/handlers/multisigCallHandler.ts

```diff
@@ -34,4 +47,117 @@ export async function handleMultisigCall(
+
+  const blockCreated = timepoint ? formatStringToNumber(timepoint.height) : getBlockCreated(extrinsic);
+  const indexCreated = timepoint ? formatStringToNumber(timepoint.index) : getIndexCreated(extrinsic);
+
+  const [existingOperation] = await MultisigOperation.getByFields([
+    ['callHash', '=', callHash],
+    ['blockCreated', '=', blockCreated],
+    ['indexCreated', '=', indexCreated],
+    ['address', '=', multisigAddress]
+  ], { limit: 1 });
+
+  const operationId = existingOperation?.id || generateOperationId(callHash, multisigAddress, blockCreated, indexCreated);
+
+  const newOperation = await MultisigOperation.create({
+    id: operationId,
+    callHash,
+    status: OperationStatus.pending,
+    address: multisigAddress,
+    deposit: 1,
+    depositor: extrinsic.extrinsic.signer.toHex(),
+    blockCreated,
+    indexCreated,
+    callData: call?.toHex(),
+    ...(call ? call.toHuman() as {} : {}),
+    timestamp: timestamp(extrinsic.block)
+  });
+
+  await newOperation.save();
+  return newOperation;
+}
+
+async function updateOperationStatus(operation: MultisigOperation, status: OperationStatus) {
+  const updatedOperation = await MultisigOperation.create({
+    ...operation,
+    status,
+  });
+  await updatedOperation.save();
+  return updatedOperation;
+}
+
+async function createMultisigEvent(extrinsic: SubstrateExtrinsic, operation: MultisigOperation, status: EventStatus) {
+  const signer = extrinsic.extrinsic.signer.toString();
+
+  await MultisigEvent.create({
+    id: generateEventId(operation.id, signer, status),
+    address: signer,
+    status,
+    blockCreated: getBlockCreated(extrinsic),
+    indexCreated: getIndexCreated(extrinsic),
+    multisigId: operation.id,
+    timestamp: timestamp(extrinsic.block)
+  }).save();
+}
+
+export async function handleApproveMultisigCall(call: VisitedCall): Promise<void> {
+  const operation = await getTransaction(call.extrinsic);
+  if (!operation) return;
+
+  await createMultisigEvent(call.extrinsic, operation, EventStatus.approve);
+
+  const finalEvent = call.events.find(e => e.method === 'MultisigExecuted');
+  if (!finalEvent) return;
+
+  const resultIndex = finalEvent.data.names?.indexOf('result');
+  if (resultIndex === undefined || resultIndex === -1) return;
+
+  const result = finalEvent.data[resultIndex] as DispatchResult;
+  const status = result.isOk ? OperationStatus.executed : OperationStatus.error;
+  await updateOperationStatus(operation, status);
+}
+
+export async function handleCancelMultisigCall(call: VisitedCall): Promise<void> {
+  const { args } = call.extrinsic.extrinsic.method.toHuman() as unknown as CancelMultisigArgs;
+  const { threshold, other_signatories, timepoint, call_hash: callHash } = args;
+
+  const signer = call.extrinsic.extrinsic.signer.toString();
+  const multisigAddress = createKeyMultiAddress([...other_signatories, signer], threshold);
```

Comment:
There is no need to manually do it here. `VisitedCall` already contains `origin` field that will be

1. Proxied - for `proxy` calls
2. Mutlisig - for `multisig` calls

And it work with any level of nestedness